src/custom/priceApi/matcha-0x/index.ts

```diff
@@ -0,0 +1,137 @@
+import { getTokensFromMarket } from 'utils/misc'
+import { getValidParams, PriceInformation, PriceQuoteParams } from 'priceApi/price'
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+interface GetSwapQuoteResponseLiquiditySource {
+  name: string
+  proportion: string
+  intermediateToken?: string
+  hops?: string[]
+}
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+interface MatchaBaseQuote {
+  chainId: ChainId
+  price: string
+  buyAmount: string
+  sellAmount: string
+  sources: GetSwapQuoteResponseLiquiditySource[]
+  gasPrice: string
+  estimatedGas: string
+  sellTokenToEthRate: string
+  buyTokenToEthRate: string
+  protocolFee: string
+  minimumProtocolFee: string
+  allowanceTarget?: string
+}
+
+// https://github.com/0xProject/0x-api/blob/8c4cc7bb8d4fa06a220b7dfd5784361c05daa92a/src/types.ts#L229
+export interface MatchaPriceQuote extends MatchaBaseQuote {
+  sellTokenAddress: string
+  buyTokenAddress: string
+  value: string
+  gas: string
+}
+
+function getMatchaChainId(chainId: ChainId): NetworkID | null {
+  switch (chainId) {
+    // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+    // See https://0x.org/docs/api#introduction
+    case ChainId.MAINNET:
+    case ChainId.ROPSTEN:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+function getApiUrl(): Partial<Record<ChainId, string>> {
+  // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+  // See https://0x.org/docs/api#introduction
+  return {
+    [ChainId.MAINNET]: 'https://api.0x.org/swap',
+    [ChainId.ROPSTEN]: 'https://ropsten.api.0x.org/swap',
+  }
+}
+
+// Defaults
+const API_NAME = 'Matcha(0x)'
+const API_BASE_URL = getApiUrl()
+const API_VERSION = 'v1'
+const DEFAULT_HEADERS = {
+  'Content-Type': 'application/json',
+}
+// GPV2Settlement
+// https://etherscan.io/address/0x9008d19f58aabd9ed0d60971565aa8510560ab41
+const AFFILIATE_ADDRESS = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41'
+const EXCLUDED_SOURCES = ''
+const MATCHA_DEFAULT_OPTIONS = `&affiliateAddress=${AFFILIATE_ADDRESS}&excludedSources=${EXCLUDED_SOURCES}`
+
+function _getApiBaseUrl(chainId: ChainId): string {
+  const baseUrl = API_BASE_URL[chainId]
+
+  if (!baseUrl) {
+    throw new Error(`Unsupported Network. The ${API_NAME} API is not deployed in the Network ${chainId}`)
+  } else {
+    return baseUrl + '/' + API_VERSION
+  }
+}
+
+function _fetch(chainId: ChainId, url: string, method: 'GET' | 'POST' | 'DELETE', data?: any): Promise<Response> {
+  const baseUrl = _getApiBaseUrl(chainId)
+  return fetch(baseUrl + url, {
+    headers: DEFAULT_HEADERS,
+    method,
+    body: data !== undefined ? JSON.stringify(data) : data,
+  })
+}
+
+// TODO: consider making these _get/_delete/_post etc reusable across apis
+function _get(chainId: ChainId, url: string): Promise<Response> {
+  return _fetch(chainId, url, 'GET')
+}
```

Comment:
>  consider making these _get/_delete/_post etc reusable across apis

Came here to say that.
Since you are refactoring already, too much trouble to put `_fetch`/`_get`/etc it into `priceApi/index` or something?